cortado-core/cortado/src/main/java/edu/utexas/cs/utopia/cortado/CmdLine.java

```diff
@@ -0,0 +1,587 @@
+package edu.utexas.cs.utopia.cortado;
+
+import com.microsoft.z3.Version;
+import edu.utexas.cs.utopia.cortado.util.logging.CortadoProfiler;
+import edu.utexas.cs.utopia.cortado.util.sat.maxsat.WeightedPartialMaxSatSolverFactory;
+import edu.utexas.cs.utopia.cortado.util.sat.maxsat.Z3WeightedPartialMaxSatSolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Copied and Modified from edu.utexas.cs.utopia.cfpchecker.CmdLine by kferles
+ * <p>
+ * Holds command line arguments for this project
+ */
+class CmdLine
+{
+    private static final Logger log = LoggerFactory.getLogger(CmdLine.class.getName());
+    // classes to process
+    private final List<String> targetClasses = new ArrayList<>();
+    // silence warnings?
+    // print .dot files of control flow?
+    private boolean fragPartToDot = false;
+    private boolean fragPartToSmallDot = false;
+    private String dotDir;
+    // is user requesting help screen?
+    private boolean isHelp = false;
+    // signal inference options
+    private String signalInfAlgorithm = "expresso";
+    private final List<String> signalInferenceOpts = Arrays.asList("annotations", "expresso");
+    private String flipPredSigOpt = "unconditional-sigs";
+    private final List<String> flipPredSigOptOpts = Arrays.asList("never", "unconditional-sigs", "always");
+    private boolean cascadingBcasts = true;
+    // which rw set analysis to use
+    private String rwSetAnalysis = "soot";
+    private final List<String> rwSetAnalyses = Arrays.asList("soot", "dummy");
+    // how should we build code fragments
+    private String fragmentConstructor = "pt-based";
+    private final List<String> fragmentConstructors = Arrays.asList("annotationsFinest", "annotationsCoarse", "ccr", "statement", "pt-based");
+    private String fragmentWeightMethod = "par-op";
+    private final List<String> fragmentWeightMethods = Arrays.asList("par-op", "uniform");
+
+    private long maxSatTimeoutInMs = TimeUnit.SECONDS.toMillis(20);
+    private long expressoSMTDefaultTimeoutInMs = TimeUnit.MINUTES.toMillis(10);
+    private long cortadoSMTDefaultTimeoutInMs = TimeUnit.SECONDS.toMillis(15);
+    // lock enumeration
+    private boolean enumerateLockings = false;
+    private long lockingEnumerationTimeoutInMs = TimeUnit.SECONDS.toMillis(10);
+    private String enumerationOutputDir = null;
```

Comment:
Can we drop the enumeration options (also their relevant functions and help docs)

I think enumeration has not been run in a long time, and wasn't used much in the paper